### PR TITLE
Display full BTC deposit address in deposit flow

### PR DESCRIPTION
**Pull request by r-czajkowski**
Closes: #401 

Here we display the full BTC deposit address in the deposit flow. This PR also refactors the `CopyToClipboard` component to be more flexible so we can use the same copy logic but with different UI components.

### CODE REVIEW

File: src/pages/tBTC/Bridge/MintingCard/MakeDeposit.tsx [hunk 1]
```diff
@@ -28,14 +31,85 @@ const AddressRow: FC<{ address: string; text: string }> = ({
...
+            flex="1"
+            p="4"
+            display="flex"
+            alignSelf="stretch"
+            alignItems="center"
+          >
+            <CopyToClipboardButton />
+          </BTCAddressCard>
+        </HStack>
+      </CopyToClipboard>
```
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/MintingCard/MakeDeposit.tsx:108]: Non-blocking but would be nice to get it resolved:

The `MakeDeposit` page is broke for 768-991px width (see the timeline on the right):

image

Would be nice to fix that although I am marking it as non-blocking

**Review by michalsmiarowski — COMMENTED** [review]
Left some non-blocking comments to look at:

File: src/components/CopyToClipboard/index.tsx [hunk 2]
```diff
@@ -16,7 +16,57 @@ type CopyToClipboardProps = {
   copyTextMsg?: string
 }
 
-const CopyToClipboard: FC<CopyToClipboardProps> = ({
+type CopyToClipboardContextValue = {
+  hasCopied: boolean
+  onCopy: () => void
+  copyTextMsg: string
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply src/components/CopyToClipboard/index.tsx]: e069bb7

File: src/pages/tBTC/Bridge/MintingCard/MakeDeposit.tsx (hunk 1 above)
Comment (r-czajkowski) [reply src/pages/tBTC/Bridge/MintingCard/MakeDeposit.tsx:108]: 4fe5c99

**Review by michalsmiarowski — APPROVED** [review]
LGTM 🔥

**Pull request merged**